discovery-provider/src/tasks/users.py

```diff
@@ -225,10 +235,74 @@ def parse_user_event(
         user_record.cover_photo = None
     return user_record
 
-def get_metadata_overrides_from_ipfs(session, update_task, user_record):
+def update_user_associated_wallets(session, update_task, user_record, associated_wallets):
+    """ Updates the user associated wallets table """
+    try:
+        if not isinstance(associated_wallets, dict):
+            return
+
+        prev_user_associated_wallets_response = (
+            session.query(AssociatedWallet.wallet)
+            .filter_by(user_id=user_record.user_id, is_current=True, is_delete=False)
+            .all()
+        )
+
+        previous_wallets = [wallet for [wallet] in prev_user_associated_wallets_response]
+        added_associated_wallets = set()
+        deleted_wallets = []
+
+        session.query(AssociatedWallet).filter_by(user_id=user_record.user_id).update({ "is_current": False })
+
+        # Verify the wallet signatures and create the user id to wallet associations
+        for associated_wallet, signature in associated_wallets.items():
+            signed_wallet = recover_user_id_hash(update_task.web3, user_record.user_id, signature)
+
+            if signed_wallet == associated_wallet:
+                # Check that the wallet doesn't already exist
+                wallet_exists = session.query(AssociatedWallet).filter_by(wallet = associated_wallet, is_current=True, is_delete=False).count() > 0
```

Comment:
should this filter by user_id? just thinking if someone has multiple accounts and links the same wallet to others

Reply:
We are explicitly not allowing that a single wallet to be linked to multiple users. 
A wallet can only be linked to a single user at a time.

Reply:
is that enforced somewhere?

Reply:
enforced by this line and the check below `if not wallet_exists`